src/app/qgsmaptoolcircle2points.cpp

```diff
@@ -0,0 +1,64 @@
+/***************************************************************************
+    qgmaptoolcircle2points.cpp  -  map tool for adding circle
+    from 2 points
+    ---------------------
+    begin                : July 2017
+    copyright            : (C) 2017 by Loïc Bartoletti
+    email                : lbartoletti at tuxfamily dot org
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsmaptoolcircle2points.h"
+#include "qgsgeometryrubberband.h"
+#include "qgsmapcanvas.h"
+#include "qgspoint.h"
+#include <QMouseEvent>
+
+QgsMapToolCircle2Points::QgsMapToolCircle2Points( QgsMapToolCapture *parentTool,
+    QgsMapCanvas *canvas, CaptureMode mode )
+  : QgsMapToolAddCircle( parentTool, canvas, mode )
+{
+
+}
+
+void QgsMapToolCircle2Points::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
+{
+  QgsPoint mapPoint( e->mapPoint() );
+
+  if ( e->button() == Qt::LeftButton )
+  {
+    mPoints.append( mapPoint );
+
+    if ( !mPoints.isEmpty() )
+    {
+      if ( !mTempRubberBand )
```

Comment:
Another minor comment : these two successives `if` may be concatenated to avoid an unneeded level of indentation : 

```
if ( !mPoints.isEmpty() && !mTempRubberBand )
```
```

Reply:
done